packages/core/src/readers/MarkdownReader.ts

```diff
@@ -95,15 +95,15 @@ export class MarkdownReader implements FileReader {
     const content = await fs.readFile(file);
     const tups = this.parseTups(content);
     const results: Document[] = [];
-    for (const [header, value] of tups) {
+        for (const [header, value] of tups) {
+      const sha256 = createSHA256()
       if (header) {
-        results.push(
-          new Document({
-            text: `\n\n${header}\n${value}`,
-          }),
-        );
+        const text = `\n\n${header}\n${value}`;
+        sha256.update(text);
+        results.push(new Document({ text, id_: sha256.digest() }));
```

Comment:
@ezirmusitua great catch! The other readers are using `file` as `id_`. How about doing the same here (and appending a suffix as it seems more than one document is generated per file here)

Reply:
In my project i use pdf and markdown documents, so i use sha256 digest, may be we should use `file` in `PDFReader` too?

https://github.com/run-llama/LlamaIndexTS/blob/ca348a657053b8f3d3427029d9dfcf343ed03fe7/packages/core/src/readers/PDFReader.ts#L21

Reply:
Hash content may be costly if it's enormous. So maybe filePath or something?

Reply:
@himself65 yes, that's what I mean: using `file` as ID. @ezirmusitua feel free to send a PR - otherwise, we will for sure change this in an upcoming release